egs/formosa/s5/run.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+#
+# Copyright 2018, Yuan-Fu Liao, National Taipei University of Technology, yfliao@mail.ntut.edu.tw
+#
+# Before you run this recips, please apply, download and put or make a link of the corpus under this folder (folder name: "NER-Trs-Vol1").
+# For more detail, please check:
+# 1. Formosa Speech in the Wild (FSW) project (https://sites.google.com/speech.ntut.edu.tw/fsw/home/corpus)
+# 2. Formosa Speech Recognition Challenge (FSW) 2018 (https://sites.google.com/speech.ntut.edu.tw/fsw/home/challenge)
+
+# shell options
+set -e -o pipefail
+
+. ./cmd.sh
+. ./utils/parse_options.sh
+
+# configure number of jobs running in parallel, you should adjust these numbers according to your machines
+numjobs=20
+numjobs_test=20
+
+# Lexicon Preparation,
+echo "$0: Lexicon Preparation"
+local/prepare_dict.sh || exit 1;
+
+# Data Preparation
+echo "$0: Data Preparation"
+local/prepare_data.sh || exit 1;
+
+# Phone Sets, questions, L compilation
+echo "$0: Phone Sets, questions, L compilation Preparation"
+rm -rf data/lang
+utils/prepare_lang.sh --position-dependent-phones false data/local/dict \
+    "<SIL>" data/local/lang data/lang || exit 1;
+
+# LM training
+echo "$0: LM training"
+rm -rf data/local/lm/3gram-mincount
+local/train_lms.sh || exit 1;
+
+# G compilation, check LG composition
+echo "$0: G compilation, check LG composition"
+utils/format_lm.sh data/lang data/local/lm/3gram-mincount/lm_unpruned.gz \
+    data/local/dict/lexicon.txt data/lang_test || exit 1;
+
+# Now make MFCC plus pitch features.
+# mfccdir should be some place with a largish disk where you
+# want to store MFCC features.
+echo "$0: naking mfccs"
+mfccdir=mfcc
+
+for x in train test; do
+  steps/make_mfcc_pitch.sh --cmd "$train_cmd" --nj $numjobs data/$x exp/make_mfcc/$x $mfccdir || exit 1;
+  steps/compute_cmvn_stats.sh data/$x exp/make_mfcc/$x $mfccdir || exit 1;
+  utils/fix_data_dir.sh data/$x || exit 1;
+done
+
+# Make some small data subsets for early system-build stages.  Note, there are 29k
+# utterances in the train_clean_100 directory which has 100 hours of data.
+# For the monophone stages we select the shortest utterances, which should make it
+# easier to align the data from a flat start.
+
+echo "$0: make training subsets"
+utils/subset_data_dir.sh --shortest	data/train   3000  data/train_mono
```

Comment:
I think you are overdoing it a bit with these data subsets.  I suggest to use a subset for mono, but use all the data for the triphone stages.  They are pretty fast.